test/src/unit-max-tile-size.cc

```diff
@@ -0,0 +1,1098 @@
+  cppapi_max = cpp_get_fragments_max_in_memory_tile_size(main_array_name_);
+  CHECK(capi_max == 16);
+  CHECK(cppapi_max == capi_max);
+
+  // Consolidate.
+  tiledb::Config config;
+  config.set("sm.consolidation.buffer_size", "10000");
+  tiledb::Array::consolidate(ctx_, main_array_name_, &config);
+  capi_max = c_get_fragments_max_in_memory_tile_size(main_array_name_);
+  cppapi_max = cpp_get_fragments_max_in_memory_tile_size(main_array_name_);
+
+  // Now '24' after consolidate(), data tiles pick up extra overhead to support
+  // time traveling.
+  CHECK(capi_max == 24);
+  CHECK(cppapi_max == capi_max);
+
+  write_sparse_array_string_dim_int32_attr(3);
+  capi_max = c_get_fragments_max_in_memory_tile_size(main_array_name_);
+  cppapi_max = cpp_get_fragments_max_in_memory_tile_size(main_array_name_);
+  CHECK(capi_max == 24);
+  CHECK(cppapi_max == capi_max);
+  write_sparse_array_string_dim_int32_attr(1);
+  capi_max = c_get_fragments_max_in_memory_tile_size(main_array_name_);
+  cppapi_max = cpp_get_fragments_max_in_memory_tile_size(main_array_name_);
+  CHECK(capi_max == 24);
+  CHECK(cppapi_max == capi_max);
+
+  // Consolidate.
+  tiledb::Array::consolidate(ctx_, main_array_name_, &config);
+  capi_max = c_get_fragments_max_in_memory_tile_size(main_array_name_);
+  cppapi_max = cpp_get_fragments_max_in_memory_tile_size(main_array_name_);
+
+  // Changed, data tiles pick up extra overhead to support time traveling.
+  CHECK(capi_max == 56);
+  CHECK(cppapi_max == capi_max);
+
+  // Sparse/dense vacuum/consolidate semantics differ,
+  // sparse retains any old data and time travel overhead.
+  // Vacuum.
+  tiledb::Array::vacuum(ctx_, main_array_name_);
+  capi_max = c_get_fragments_max_in_memory_tile_size(main_array_name_);
+  cppapi_max = cpp_get_fragments_max_in_memory_tile_size(main_array_name_);
+
+  // No change, vacuuming sparse does not remove the earlier data.
+  CHECK(capi_max == 56);
+  CHECK(cppapi_max == capi_max);
+
+  // Secondary attempt still retains the overhead.
+  tiledb::Array::consolidate(ctx_, main_array_name_, &config);
+  capi_max = c_get_fragments_max_in_memory_tile_size(main_array_name_);
+  cppapi_max = cpp_get_fragments_max_in_memory_tile_size(main_array_name_);
+
+  // No change after consolidate(), extra
+  // overhead retained across multiple consolidate()s.
+  CHECK(capi_max == 56);
+  CHECK(cppapi_max == capi_max);
+
+  // Vacuum.
+  tiledb::Array::vacuum(ctx_, main_array_name_);
+  capi_max = c_get_fragments_max_in_memory_tile_size(main_array_name_);
+  cppapi_max = cpp_get_fragments_max_in_memory_tile_size(main_array_name_);
+
+  // No change, vacuuming sparse does not remove the earlier data, even after a
+  // secondary consolidate against previously vacuumed data.
+  CHECK(capi_max == 56);
+  CHECK(cppapi_max == capi_max);
+}
+
+TEST_CASE_METHOD(
+    MaxTileSizeFx,
+    "C++ API: Max fragment size, dense create/evolve, fixed dim/attr",
+    "[capi][max-tile-size][dense][evolve][consolidate]") {
```

Comment:
In all the tests above the comment define what the intent is and it makes reading the previous tests cases so easy. Could we add similar comments here?

Reply:
Made a pass through adding additional comments.